clients/src/main/java/org/apache/kafka/common/protocol/Errors.java

```diff
@@ -587,8 +603,8 @@ public static Errors forCode(short code) {
         if (error != null) {
             return error;
         } else {
-            log.warn("Unexpected error code: {}.", code);
-            return UNKNOWN;
+            log.warn("Unknown error code: {}.", code);
+            return UNKNOWN_RETRIABLE;
```

Comment:
Yes that is the plan and it is actually implemented here. Note that `Errors.forException(Throwable t)` will return `Errors.UNKNOWN` if the given exception is not listed in `Errors.java`. And `Errors.forCode(short code)` will return `Errors.UNKNOWN_RETRIABLE` is the error code is out of the range of existing error codes listed in `Errors.java`. 

Our server code is not expected to call `Errors.forCode(short code)` with a not-listed error code. The client may call `Errors.forCode(short code)` with a not-listed error code if the client library version is smaller than the server library version and the server library has a new error code. In this case we want `Errors.forCode(short code)` to return a retriable exception so that producer can re-send the message after metadata update. Does this make sense?